gc3libs/utils.py

```diff
@@ -2109,40 +1883,17 @@ def to_bytes(s):
       1048576
 
     """
-    last = -1
-    unit = s[last].lower()
-    if unit.isdigit():
-        # `s` is a integral number
+    # get the multiplier
+    k = 1024 if 'i' in s.lower() else 1000
+    # strip bad stuff away now
+    s = s.lower().strip(' ib')
+    # if it is just a number, return as int
+    if s.isdigit():
         return int(s)
-    if unit == 'b':
-        # ignore the the 'b' or 'B' suffix
-        last -= 1
-        unit = s[last].lower()
-    if unit == 'i':
-        k = 1024
-        last -= 1
-        unit = s[last].lower()
-    else:
-        k = 1000
-    # convert the substring of `s` that does not include the suffix
-    if unit.isdigit():
-        return int(s[0:(last + 1)])
-    if unit == 'k':
-        return int(float(s[0:last]) * k)
-    if unit == 'm':
-        return int(float(s[0:last]) * k * k)
-    if unit == 'g':
-        return int(float(s[0:last]) * k * k * k)
-    if unit == 't':
-        return int(float(s[0:last]) * k * k * k * k)
-    if unit == 'p':
-        return int(float(s[0:last]) * k * k * k * k * k)
-    if unit == 'e':
-        return int(float(s[0:last]) * k * k * k * k * k * k)
-    if unit == 'z':
-        return int(float(s[0:last]) * k * k * k * k * k * k * k)
-    if unit == 'y':
-        return int(float(s[0:last]) * k * k * k * k * k * k * k * k)
+    num, unit = s[:-1], s[-1]
+    # map suffix to pow multipliers
+    mapped = {l: i for i, l in enumerate('kmgtpezy', start=1)}
```

Comment:
dict literal syntax not compatible with Py2.6

Reply:
Also, not that this function is performance critical, but since the `mapped` dict is not going to change, why not define it at load-time?

```python
_units_scale = dict((letter, exponent) for exponent, letter in enumerate('kmgtpezy', start=1))

def to_bytes(...):
  # ...
  return int(num) * (k ** _units_scale[unit])
```